src/contracts/actions-builders/EverclearTokenStake.sol

```diff
@@ -0,0 +1,151 @@
+   * @param _next The NEXT contract address
+   * @param _clear The CLEAR contract address
+   * @param _lockTime The lock time
+   */
+  constructor(
+    address _vestingEscrow,
+    address _vestingWallet,
+    address _spokeBridge,
+    address _clearLockbox,
+    address _next,
+    address _clear,
+    uint256 _lockTime
+  ) {
+    VESTING_ESCROW = IVestingEscrow(_vestingEscrow);
+    VESTING_WALLET = IVestingWallet(_vestingWallet);
+    SPOKE_BRIDGE = ISpokeBridge(_spokeBridge);
+    CLEAR_LOCKBOX = IxERC20Lockbox(_clearLockbox);
+    NEXT = IERC20(_next);
+    CLEAR = IERC20(_clear);
+    LOCK_TIME = _lockTime;
+  }
+
+  // ~~~ ACTIONS METHODS ~~~
+
+  /// @inheritdoc IActionsBuilder
+  function getActions() external view returns (Action[] memory _actions) {
+    _actions = new Action[](6);
+
+    // NOTE: since this is a view function and does not update state, we need to calculate
+    // how much tokens are transferred when calling release() on step 2.
+
+    // First get the unclaimed amount
+    uint256 _unclaimed = VESTING_ESCROW.unclaimed();
+    // Get current balance of NEXT
+    uint256 _nextBalance = NEXT.balanceOf(address(VESTING_WALLET));
+    // Get the releaseable amount (same behaviour as releaseable() on VESTING_WALLET)
+    uint256 _amountReleasable = VESTING_WALLET.vestedAmount(uint64(block.timestamp)) - VESTING_WALLET.released();
+    uint256 _nextBalanceAfterRelease = _nextBalance + _unclaimed;
+    uint256 _amountToBeReleased =
+      _nextBalanceAfterRelease < _amountReleasable ? _nextBalanceAfterRelease : _amountReleasable;
+
+    // 1) Claim
+    _actions[0] = Action({
+      target: address(VESTING_WALLET),
+      data: abi.encodeCall(IVestingWallet.claim, (address(VESTING_ESCROW))),
+      value: 0
+    });
+
+    // 2) Release
+    _actions[1] = Action({target: address(VESTING_WALLET), data: abi.encodeCall(IVestingWallet.release, ()), value: 0});
+
+    // 3) Approve
+    _actions[2] = Action({
+      target: address(NEXT),
+      data: abi.encodeCall(IERC20.approve, (address(CLEAR_LOCKBOX), _amountToBeReleased)),
+      value: 0
+    });
+
+    // 4) Deposit
+    _actions[3] = Action({
+      target: address(CLEAR_LOCKBOX),
+      data: abi.encodeCall(IxERC20Lockbox.deposit, (_amountToBeReleased)),
+      value: 0
+    });
+
+    // 5) Approve
+    _actions[4] = Action({
+      target: address(CLEAR),
+      data: abi.encodeCall(IERC20.approve, (address(SPOKE_BRIDGE), _amountToBeReleased)),
+      value: 0
+    });
+
+    // 6) Increase lock position
+
+    uint256 _gasLimit = 500_000;
+
+    // NOTE: expiry % 7 days must be 0
+    uint128 _lockTime = uint128(block.timestamp + LOCK_TIME);
+    _lockTime = (_lockTime / 1 weeks) * 1 weeks;
+
+    // NOTE: get the fee from the gateway
+    uint256 _value = IGateway(SPOKE_BRIDGE.gateway()).quoteMessage(
+      SPOKE_BRIDGE.EVERCLEAR_ID(), abi.encode(2, _getSenderFromEntrypoint(), _amountToBeReleased, _lockTime), _gasLimit
+    );
```

Comment:
mhmmm, we're calculating here the fee amount, but it may be different than at the time of execution, which would be bad, i don't know if it's possible to do, yikes